### PR TITLE
Patch to Oga Upstream to fix "Oga Parser doesn't handle dots in XML T…

### PR DESCRIPTION
**Pull request by rubyjedi**
Original problem description is at rubyjedi/soap4r#7

Mahalo for your time in reviewing this fix.

### CODE REVIEW

**Comment by abotalov** [comment]
There are a number of symbols that might be contained in tag name:
http://www.w3.org/TR/xml/#NT-Name

Only alphanumeric ASCII characters are allowed in HTML, however:
http://www.w3.org/TR/html51/syntax.html#syntax-tag-name

**Comment by YorickPeterse** [comment]
Looking at the spec it seems XML also allows Unicode in names (I always assumed it was ASCII only). For Unicode the lexer can use an approach similar to https://github.com/YorickPeterse/ruby-ll/blob/09249143588817f355ac6e3ddbc8738fda738acc/lib/ll/lexer.rl#L164-L166, although the rules would have to be changed so they _don't_ allow names starting with a number. If this is too daunting I can take care of this.

As an aside, a small test for supporting dots would be appreciated, it can be put in https://github.com/YorickPeterse/oga/blob/master/spec/oga/xml/lexer/elements_spec.rb.

**Comment by rubyjedi** [comment]
'Happy to put in a spec to test for inline dots -- should show up alongside this P/R in a few hours.

As for refactoring the lexer, I'd much prefer leaving that to your expertise. Thanks for all your great work, guys! Much appreciated!

File: spec/oga/xml/lexer/elements_spec.rb [hunk 1]
```diff
@@ -307,4 +307,19 @@
       ]
     end
   end
+  
+  describe 'elements with inline dots' do
+    it 'lexes an element with inline dots' do
+        lex('<SOAP..TestMapping..MappablePerson env:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" />').should == [
```
Comment (YorickPeterse) [inline spec/oga/xml/lexer/elements_spec.rb]: This spec can be trimmed down to:

```ruby
it 'lexes an element with a dot in the name' do
  lex('<foo.bar />').should == [...]
end
```

The extra describe isn't really needed, nor is the attribute value. If attribute names are also allowed to contain dots this should be tested in a separate `it` block.

**Comment by rubyjedi** [comment]
.. remind me not to push code so late at night, nor early in the morning before coffee. :o)

This spec update should be much more appropriate.

**Comment by YorickPeterse** [comment]
Rebased this into commits https://github.com/YorickPeterse/oga/compare/71960fff87da633dcab863002a461fbf7d4c5738...139985612b3a37319f01a2e1f97ad00c23f64c54 (due to a bunch of changes on the master branch). Thanks!

**Pull request closed without merging**